File: snippets/react-native/closed_channel.ts
import {
  prepareRedeemOnchainFunds,
  redeemOnchainFunds
} from '@breeztech/react-native-breez-sdk'

const examplePrepareRedeemOnchainFunds = async (feeRate: number) => {
  // ANCHOR: prepare-redeem-onchain-funds
  try {
    const toAddress = 'bc1..'
    const satPerVbyte = feeRate
    const prepareRedeemOnchainFundsResp = await prepareRedeemOnchainFunds({ toAddress, satPerVbyte })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: prepare-redeem-onchain-funds
}

const exampleRedeemOnchainFunds = async (satPerVbyte: number, toAddress: string) => {
  // ANCHOR: redeem-onchain-funds
  try {
    const redeemOnchainFundsResp = await redeemOnchainFunds({ toAddress, satPerVbyte })
  } catch (err) {
    console.error(err)
  }
  // ANCHOR_END: redeem-onchain-funds
}

File: snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/ClosedChannel.kt
package com.example.kotlinmpplib

import breez_sdk.*
class ClosedChannel {
    fun prepareRedeemOnchainFunds(sdk: BlockingBreezServices, feeRate: UInt) {
        // ANCHOR: prepare-redeem-onchain-funds 
         val satPerVbyte = feeRate
         var destinationAddress = "bc1.."
         try {
            val req = PrepareRedeemOnchainFundsRequest(destinationAddress, satPerVbyte)
            val prepareRedeemOnchainFundsResp = sdk.prepareRedeemOnchainFunds(req)
         }
         catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: prepare-redeem-onchain-funds 
    }

    fun redeemOnchainFunds(sdk: BlockingBreezServices, satPerVbyte: UInt, toAddress: String) {
        // ANCHOR: redeem-onchain-funds 
         try {
            val req = RedeemOnchainFundsRequest(toAddress, satPerVbyte)
            val redeemOnchainFundsResp = sdk.redeemOnchainFunds(req)
         }
         catch (e: Exception) {
            // handle error
        }
        // ANCHOR_END: redeem-onchain-funds
    }
}